src/rawtoaces_core/rawtoaces_core.cpp

```diff
@@ -217,227 +218,201 @@ void scaleLSC( const SpectralData &camera, SpectralData &illuminant )
     illuminant_spectrum *= scale;
 }
 
-//	=====================================================================
-//	Load the Camera Sensitivty data
-//
-//	inputs:
-//		const std::string &: path to the camera sensitivity file
-//      const std::string &: camera maker  (from libraw)
-//      const std::string &: camera model  (from libraw)
-//
-//	outputs:
-//		int: If successfully parsed, _cameraSpst will be filled and return 1;
-//               Otherwise, return 0
+std::vector<std::string>
+SpectralSolver::collect_data_files( const std::string &type ) const
+{
+    std::vector<std::string> result;
+
+    for ( const auto &directory: _search_path )
+    {
+        if ( std::filesystem::is_directory( directory ) )
+        {
+            std::filesystem::path type_path( directory );
+            type_path.append( type );
+            if ( std::filesystem::exists( type_path ) )
+            {
+                auto it = std::filesystem::directory_iterator( type_path );
+                for ( auto filename2: it )
+                {
+                    auto p = filename2.path();
+                    if ( filename2.path().extension() == ".json" )
+                    {
+                        result.push_back( filename2.path().string() );
+                    }
+                }
```

Comment:
```suggestion
                for ( auto filename: it )
                {
                    auto p = filename.path();
                    if ( filename.path().extension() == ".json" )
                    {
                        result.push_back( filename.path().string() );
                    }
                }
```
I can see you've cut/paste that from other place. But why not improve it just a touch while we are here.